src/server/error.ts

```diff
@@ -0,0 +1,39 @@
+export enum ServerErrorCode {
+  CONTINUE = 100,
+
+  OK = 200,
+  CREATED = 201,
+  ACCEPTED = 202,
+  NO_CONTENT = 204,
+
+  BAD_REQUEST = 400,
+  UNAUTHORIZED = 401,
+  FORBIDDEN = 403,
+  NOT_FOUND = 404,
+  CONFLICT = 409,
+
+  INTERNAL_SERVER_ERROR = 500,
+  NOT_IMPLEMENTED = 501,
+  BAD_GATEWAY = 502,
+  SERVICE_UNAVAILABLE = 503,
+  GATEWAY_TIMEOUT = 504,
+}
+
+export class ServerError extends Error {
+  key: string
+  params: any
+  statusCode: ServerErrorCode
+
+  constructor(key: string, params?: any, statusCode: ServerErrorCode = ServerErrorCode.INTERNAL_SERVER_ERROR) {
```

Comment:
Should `statusCode` be optional?

Reply:
it is optional. it takes  ServerErrorCode.INTERNAL_SERVER_ERROR if not passed.
or I misunderstood the comment?